### ISSUE TITLE
Removing Gemfile.lock

### ISSUE THREAD

**Issue by fpigerre** [2factorauth/twofactorauth#1336]
At the moment, when installing gems from TwoFactorAuth's gemfile on some windows platforms, the following error occurs, stopping all gems from being installed.

This is due to the `parallel` gem being included in TwoFactorAuth's `Gemfile.lock` file.
Removing the Gemfile.lock file permanently from the repository and adding it to the `.gitignore` file would fix this issue and a `Gemfile.lock` file could instead be generated on an individual basis when running `bundle install`.
It is possible, however, that the `Gemfile.lock` file provides certain benefits to those making changes to TwoFactorAuth's codebase, such as normalizing development environments.

```
> bundle install
Fetching gem metadata from https://rubygems.org/...........
Fetching version metadata from https://rubygems.org/...
Fetching dependency metadata from https://rubygems.org/..
Resolving dependencies...
Could not find parallel-1.3.3 in any of the sources
```

Are there any reasons for which the `Gemfile.lock` file should remain in the repository?

Thanks,
psgs :palm_tree:

**Comment by mxxcon** [comment]
According to http://bundler.io/v1.10/rationale.html

> After developing your application for a while, check in the application together with the `Gemfile` and `Gemfile.lock` snapshot. Now, your repository has a record of the exact versions of all of the gems that you used the last time you know for sure that the application worked. Keep in mind that while your `Gemfile` lists only three gems (with varying degrees of version strictness), your application depends on dozens of gems, once you take into consideration all of the implicit requirements of the gems you depend on.
> 
> This is important: **the Gemfile.lock makes your application a single package of both your own code and the third-party code it ran the last time you know for sure that everything worked.** Specifying exact versions of the third-party code you depend on in your Gemfile would not provide the same guarantee, because gems usually declare a range of versions for their dependencies. 

So from that I think that `Gemfile` should have gems that your app uses directly, whereas `Gemfile.lock` it will list everything, including dependencies of dependencies with exact build numbers, ensuring anybody else that clones that repo will get exactly the same gem builds.

**Comment by jamcat22** [comment]
Why don't we remove versions in the lock file then?

**Comment by mxxcon** [comment]
We can if things won't break...

**Comment by jamcat22** [comment]
Closed via #1351.
